### PR TITLE
Part Design: Improved chamfer behaviour when faces are selected and non equal chamfer types are used

### PR DESCRIPTION
**Pull request by troyp76**
**Draft Pull Request for developer review**

Improved Part Design chamfer behaviour when faces are selected and non equal chamfer types are used. (Distance Distance or Distance Angle)
See  [Forum discussion](https://forum.freecadweb.org/viewtopic.php?f=19&t=62084) for background.

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [X]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [X]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [X]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [X]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [X]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [X]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the [FreeCAD bug tracker issue number](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) in case a particular commit solves or is related to an existing issue on the tracker. Ex: `Draft: fix typos - fixes #0004805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

File: src/Mod/PartDesign/App/FeatureChamfer.cpp [hunk 1]
```diff
@@ -115,7 +115,8 @@ App::DocumentObjectExecReturn *Chamfer::execute(void)
     }
 
     std::vector<std::string> SubNames = std::vector<std::string>(Base.getSubValues());
-    getContiniusEdges(TopShape, SubNames);
+    std::vector<std::string> FaceNames;
+    getContiniusEdges(TopShape, SubNames, FaceNames);
```
Comment (plaes) [inline src/Mod/PartDesign/App/FeatureChamfer.cpp:119]: Typo: `s/Continius/Continous`

Comment (troyp76) [reply src/Mod/PartDesign/App/FeatureChamfer.cpp:119]: Thanks for the review. This typo was in the original code see #5031 where luzpaz has fixed this now.

I will rebase this PR with his fix soon.

**Comment by chennes** [comment]
Did the line endings change here? If I don't hide whitespace changes GitHub thinks the whole file changed. (Might be GH wonkiness, though)

**Comment by troyp76** [comment]
@chennes 
Sorry I don't know what happened, I am still trying to work out how GH works.
It prompted me to resolve the conflicts after #5031 was merged. 

I then did these steps:

1. Click on the resolve conflict button.
2. Changed the function name to match the master branch
3. Removed the conflict makers to enable the "Mark Resolved" button
4. Pressed the "Mark Resolved" button

I didn't touch anything else so I assume GH touched the rest of the file.

Any advise going forward? 
Should I / Can I delete this commit and redo the resolution offline to clean it up?

**Comment by chennes** [comment]
That workflow sounds totally fine to me, but I admit I never use GitHub's online editor. I can try to take a closer look at it tomorrow.

File: src/Mod/PartDesign/App/FeatureChamfer.cpp (hunk 1 above)
Comment (0penBrain) [reply src/Mod/PartDesign/App/FeatureChamfer.cpp]: @plaes "continUous" replacing a typo by another isn't of great help generally.

**Comment by 0penBrain** [comment]
It would also be good to squash the commits. So probably fix the line ending at the same time locally then force push. ;)

**Comment by troyp76** [comment]
@0penBrain  @chennes 
I resaved the files then used GitHub Desktop to squash the commits and forced pushed it but it doesn't look like it changed anything.

Anyway I have learnt my lesson, don't do merges / conflict resolutions using the GH web interface. :)

Is this PR ready to come out of draft now?

**Comment by 0penBrain** [comment]
@troyp76 Looks you didn't actually squash anything as your PR is now 6 commits long !!!
Could you have a look to squash into a single commit ?
If you think your PR is mature enough, you can remove the draft status. I'll try to have a look a bit later.

**Comment by troyp76** [comment]
Squashing using GitHub Desktop looks like it completes but doesn't actual change anything.
Looking at the GitHub documentation it seems Squashing is not available for draft PR's.
So I will convert it and try again.

**Comment by 0penBrain** [comment]
@troyp76 Indeed, now it's 7 commits :rofl: I can't help as only know command line.

Regarding the line ending characters, you can fix this using your editor if it supports it, or use dos2unix/unix2dos commands directly on the files. On Windows, you can get implementations of these commands for example on the GoW project : https://github.com/bmatzelle/gow/tree/master/bin

**Comment by troyp76** [comment]
Hi all,
I worked out how to do the squash and force push from the command line but now the PR is just 1 big lot of changes which is not helpful for reviewing. :(

Going to call it a night and the close this Draft PR and delete the branch and start fresh ones tomorrow.

It's been a learning experience coming from SVN, thanks for the suggestions along the way.

**Comment by 0penBrain** [comment]
@troyp76 Looks like you squashed ten if not hundred of commits including a lot already in history...

**Comment by luzpaz** [comment]
Yea, the git learning curve is real. @troyp76 thanks for persevering :muscle:

**Pull request closed without merging**